sympy/concrete/products.py

```diff
@@ -399,6 +399,13 @@ def _eval_product_direct(self, term, limits):
         (k, a, n) = limits
         return Mul(*[term.subs(k, a + i) for i in range(n - a + 1)])
 
+    def _eval_derivative(self, x):
+        from sympy import Sum
+        limits = list(self.limits)
+        if any(x in limit[1:].free_symbols for limit in reversed(limits)):
```

Comment:
Do the limits need to be reversed?

Reply:
There is no specific reason, I just did so thinking that in case of multiple limits, the chances will be as the differentiating variable can be present towards the outer-side of the limits rather than in the inward limits argument. Just did it to make it faster, but it looks too abstract. 
I will remove the `reversed`.
Thanks.